app_launcher.py

```diff
@@ -31,6 +32,12 @@ def main(input_json_path):
 
     # Change environment variables
     env = data.get("env") or {}
+
+    # Add AYON_PID_FILE environment variable if pid_file is specified
+    pid_file_path = data.get("pid_file")
+    if pid_file_path and "AYON_PID_FILE" not in env:
```

Comment:
If it's passed in json then it should be set no matter what...
```suggestion
    if pid_file_path:
```

Reply:
I don't think so. I consider env vars if set as last resort override. So if you for whatever reason already have that set, it should take precedence.